### ISSUE TITLE
TableReport outliers improvement?

### ISSUE THREAD

**Issue by Vincent-Maladiere** [skrub-data/skrub#1167]
### Describe the bug

I don't think this is a bug, I'm just creating this issue so that we can observe a different behaviour I just saw.

Here is a series where the outlier detection is slightly off. This is an integer series comprised of {0, 80, 100}. Here, the values are actually categorical because they just distinguish states, and the choice of the state values 0 and 80 is arbitrary (it could have been 0, 1, 2, but for some reason it's 0, 80, 100).

Unless we tune the heuristic a little bit to avoid counting outliers when the cardinality is "very low", I don't see an easy improvement.

WDYT?

![Screenshot 2024-11-29 at 09 42 52](https://github.com/user-attachments/assets/2a6824c5-6f3e-43c2-96b0-517c12c15f00)

The column dataset:

[state.csv](https://github.com/user-attachments/files/17956631/state.csv)

### Steps/Code to Reproduce

```python
import pandas as pd
from skrub import TableReport

df = pd.read_csv("state.csv")
TableReport(df)
```

### Expected Results

No outliers

### Actual Results

Some outliers

### Versions

```shell
0.4.0 :)))
```

**Comment by jeromedockes** [comment]
here is another example from the "titanic" dataset

![screenshot_2024-11-29T10:43:55+01:00](https://github.com/user-attachments/assets/d1c153eb-71b6-4917-b4a6-4aa492a82087)

**Comment by Vincent-Maladiere** [comment]
Yes, that's tricky I agree. The best scenario would be for the user to notice that and convert to string or category dtypes. The main difference between my screenshot and yours is that on mine the outlier segment is bigger than the category displayed on the left, so the outliers are not really outliers, if that makes sense?

> what we would like in this case is to realize that the actual values don't matter and treat the variable as categorical as you say. but I'm not sure that its' reasonable to assume that is the case whenever there are few unique values 🤔

Let's wait a bit to get feedback and decide about this.
